### PR TITLE
Use ServiceLoader to load JSON-B Provider if none passed in.

### PR DESCRIPTION
**Pull request by andymc12**

### CODE REVIEW

**Comment by andymc12** [comment]
#build

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/com/ibm/ws/jaxrs21/providers/json/JsonBProvider.java [hunk 1]
```diff
@@ -42,13 +48,44 @@
...
+                    // next try this classloader
+                    providers = ServiceLoader.load(JsonbProvider.class, JsonBProvider.class.getClassLoader()).iterator();
+                    if (providers.hasNext()) {
+                        return providers.next();
+                    }
+                    
+                    // not good - but maybe we're in a test environment where there will be no
+                    // need for a JSON-B provider...
+                    if (Boolean.getBoolean("com.ibm.ws.jaxrs.testing") || System.getProperty("user.dir").contains("jaxrstck")) {
+                        return null;
```
Comment (jkoehler22) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/com/ibm/ws/jaxrs21/providers/json/JsonBProvider.java]: Need to add  || System.getProperty("user.dir").contains("javaeetck")  as Hank runs the ee tck not the jaxrs tck

**Comment by jkoehler22** [comment]
Approved....see my comment.

**Comment by andymc12** [comment]
Thanks @jkoehler22! I want to leave that out (and I just removed the TCK variant of that property) because the TCK and CTS should both include a valid JSON-B provider on the class path.  I'm leaving the original "testing" hack in place because there may be situations where we want to test JAX-RS 2.1 without needing a JSON-B provider.

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
infrastructure failed... try again - #build

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
another infrastructure issue - this time a connect timeout... hopefully this #build succeeds...

**Comment by andymc12** [comment]
failures are unrelated to this change

**Pull request merged**